scripts/build.py

```diff
@@ -1649,41 +1651,53 @@ def local_smoke(self):
 #
 # END: SMOKE TEST
 # ------------------------------------------------------------
+    def get_archive_path(self):
+        return join(self.archive_path, self.channel, "files")
+
+    def get_archive_redirect_key(self, url):
+        # s3://d.defold.com/archive/channel/files/sha1/engine/* -> s3://d.defold.com/archive/sha1/engine/*
+        old_url = url.replace(self.get_archive_path(), self.archive_path)
+        u = urlparse.urlparse(old_url)
+        return u.path
+
+    def download_from_archive(self, src_path, dst_file):
+        url = join(self.get_archive_path(), src_path)
+        self.download_from_s3(dst_file, url)
+
+    def upload_to_archive(self, src_file, dst_path, create_redirect=True):
+        url = join(self.get_archive_path(), dst_path)
+        self.upload_to_s3(src_file, url)
+        if create_redirect:
+            u = urlparse.urlparse(url)
+            bucket = s3.get_bucket(u.netloc)
+            key_name = self.get_archive_redirect_key(url)
+            key = bucket.new_key(key_name)
+            key.set_redirect(url.replace("s3://", "http://"))
 
-    def download_file(self, path, url):
+    def download_from_s3(self, path, url):
         url = url.replace('\\', '/')
         self._log('Downloading %s -> %s' % (url, path))
-        self._mkdirs(os.path.dirname(path))
         u = urlparse.urlparse(url)
 
-        if u.netloc == '':
-            print("Not implemented")
-        elif u.scheme == 's3':
+        if u.scheme == 's3':
+            self._mkdirs(os.path.dirname(path))
             from boto.s3.key import Key
 
             bucket = s3.get_bucket(u.netloc)
             k = Key(bucket)
             k.key = u.path
             k.get_contents_to_filename(path)
             self._log('Downloaded %s -> %s' % (url, path))
+        else:
+            raise Exception('Unsupported url %s' % (url))
 
-    def upload_file(self, path, url):
+    def upload_to_s3(self, path, url):
         url = url.replace('\\', '/')
         self._log('Uploading %s -> %s' % (path, url))
 
         u = urlparse.urlparse(url)
 
-        if u.netloc == '':
```

Comment:
upload_file() is only used for S3 uploads. And upload using ssh+scp is not supported anymore.